app/models/filter.rb

```diff
@@ -135,6 +141,21 @@ def expire_topbar_cache(sweeper)
     role.usergroups.each { |g| g.expire_topbar_cache(sweeper) }
   end
 
+  def disable_overriding!
+    self.override = false
+    self.save!
+  end
+
+  def inherit_taxonomies
+    inherit_taxonomies! unless self.override?
```

Comment:
The reasoning is the follow:

* The method isn't used (YAGNI, dead code)
* The rest of the code cares about 'override' inconsistently. Case in point: when I uncheck 'override', taxonomies are not disassociated from the filter object (and it's fine, they don't matter at that point)
* If someone calls `inherit_taxonomies!`, they will add any checks that are needed. The `!` is there for something.
* Single responsibility - this code does more than just `inherit_taxonomies`, it's checking a flag that